src/components/TheNav.vue

```diff
@@ -146,7 +77,57 @@ export default {
   name: 'TheNav',
   data() {
     return {
-      isNavShown: false
+      isNavShown: false,
+      navItems: [
+        {
+          path: { name: 'home' },
+          activeCondition: 'home',
+          content: 'Home',
+          conditionOne: true,
+          conditionTwo: false,
```

Comment:
Is that needed? What's the purpose?

Reply:
It is, currently, because some nav items I need to be displayed regardless of the user being signed in or not. So basically I need to have `v-show = isSignedIn === true || isSignedIn === false`. And those two conditions were the only idea I had how to set it.

Reply:
This is actually what my question was originally about, since I know it looks stupid but I have no idea how can I display some nav items regardless of the user being signed in or not.

Reply:
I hope we can improve this step by step.
First thing could be naming, but I think we could remove `conditionOne` as it's always true except `sign-in` and `sign-up`. So we could check if `path.name.includes('sign') && isSignedIn`

Reply:
Now. In the template you check `conditionTwo` only when `conditionOne` is falsy

Reply:
`conditionTwo` is true in only one route where `conditionOne` is also true. So it's not needed at all IMHO

Reply:
then I'd add a `protected` property to routes which should be hidden when user is not authenticated

Reply:
Finally I'd make a computed property as suggested before:

```js
    navItems() {
      if (this.isSignedIn) {
        return this.routes.filter((route) => route.protected == null)
      }
      return this.routes.filter((route) => route.path.name.includes('sign') === false)
    }
```

Reply:
> I hope we can improve this step by step. First thing could be naming, but I think we could remove `conditionOne` as it's always true except `sign-in` and `sign-up`. So we could check if `path.name.includes('sign') && isSignedIn`

Looks like a good idea, I'll try to implement this whole routes thing tomorrow 🙃

Reply:
**Corrected**

This was a brilliant idea, I would have never come up with a solution like this, thank you 😉